src/adapter/src/session/vars.rs

```diff
@@ -266,6 +266,14 @@ static ALLOWED_CLUSTER_REPLICA_SIZES: Lazy<ServerVar<Vec<String>>> = Lazy::new(|
     description: "The allowed sizes when creating a new cluster replica (Materialize).",
 });
 
+/// Feature flag indicating whether window functions are enabled.
+static WINDOW_FUNCTIONS_ENABLED: ServerVar<bool> = ServerVar {
+    name: UncasedStr::new("window_functions_enabled"),
+    // 1 GiB
+    value: &true,
```

Comment:
Should probably default to false.
```suggestion
    value: &false,
```

Reply:
I agree, the problem is that this is not backwards compatible. @benesch what do you think?

Reply:
Ideally it would default to `true` for all existing environments, but `false` in any new environments, I think. But I think the long term nice answer here is LaunchDarkly!

Reply:
> Ideally it would default to true for all existing environments, but false in any new environments, I think. 

I agree, but it seems non-trivial to put `value: &false` statically here and at the same time achieve the above. Once we have a version of this deployed, we might design a workflow that:

1. uses `bootstrap-system-vars` to set this to `true` for all existing environments,
2. swap the default value and the `bootstrap-system-vars` value to `false` for the next deployment.

Reply:
Since we allow items that are already in the catalog to use window functions regardless of what the feature flag says, setting the default to `false` here would not break existing objects, but would only mean that users cannot create new objects that use window functions (unless we enable it for them). Which is a product decision, but might be fine?

If we go this route, we should probably still use a `true` default here for now and switch that when we have the Launch Darky integration set up and can easily toggle the feature on for our customers.

Reply:
> Since we allow items that are already in the catalog to use window functions regardless of what the feature flag says, setting the default to false here would not break existing objects, but would only mean that users cannot create new objects that use window functions (unless we enable it for them). Which is a product decision, but might be fine?

You are right. I'll change that and sync with product before we merge.